### DISCUSSION TITLE
having trouble testing utterance_end_ms

### DISCUSSION THREAD

**Discussion by davehorton** [deepgram/community#387] [General help] [answered]
I am testing utterance_end_ms thusly
```
POST /v1/listen?model=phonecall&tier=nova&language=en-US&smart_format=true&no_delay=true&numerals=true&utterance_end_ms=1000&encoding=linear16&sample_rate=8000
```

But I presume I am doing something wrong since my websocket connection requests are not accepted, and I see a 400 error with no other info in the developer console.  An example request id is `a7246e7f-6b56-4704-bd31-1aaf0ff50dd8`.

Does anyone see what I might be doing wrong?

![image](https://github.com/deepgram/community/assets/1884973/de498a53-6655-470e-ba9e-8d8b85c79d70)

**Comment by davehorton** [comment]
is it required that `interim_results` be enabled in order for `utterance_end_ms` to be accepted?  Because I tested with `interim_results` on and now it works.....not sure if something else changed on the back end, or if my change was significant...

```
 /v1/listen?model=phonecall&tier=nova&language=en-US&smart_format=true&no_delay=true&numerals=true&interim_results=true&utterance_end_ms=1000&encoding=linear16&sample_rate=8000
```

**Reply by nikolawhallon** [reply, marked as answer]
^ exactly, the `interim_results` are part of what helps the algorithm determine when words are/aren't spoken as quickly as possible, hence why it is required

**Reply by nikolawhallon** [reply]
(I'm a little confused why it says `POST` in your logs, websockets use `GET /v1/listen`, are we reporting the wrong thing there?)

**Reply by davehorton** [reply]
I don't know either....I am definitely doing a GET
